### PR TITLE
Update PT/TF weight conversion after #24030

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Update PT/TF weight conversion due to the change in #24030.

(can do PT/Flax too in the same PR, but request a review first anyway)

### Code snippet to show issues and verify this PR's effect

(Failing for `main + nightly torch`. Pass for `PR + nightly torch` and `main/PR + stable torch`) 
```python
import transformers

from transformers import TFWav2Vec2Model
from tests.models.wav2vec2.test_modeling_tf_wav2vec2 import TFWav2Vec2ModelTest

self = TFWav2Vec2ModelTest()
self.setUp()

model_class = TFWav2Vec2Model
allow_missing_keys = False

config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()

pt_model_class_name = model_class.__name__[2:]  # Skip the "TF" at the beginning
pt_model_class = getattr(transformers, pt_model_class_name)

tf_model = model_class(config)
pt_model = pt_model_class(config)

tf_inputs_dict = self._prepare_for_class(inputs_dict, model_class)

# Check we can load pt model in tf and vice-versa with model => model functions
try:
    _tf_model = transformers.load_pytorch_model_in_tf2_model(
        tf_model, pt_model, tf_inputs=tf_inputs_dict, allow_missing_keys=allow_missing_keys
    )
except:
    _tf_model = None

try:
    _pt_model = transformers.load_tf2_model_in_pytorch_model(
        pt_model, tf_model, allow_missing_keys=allow_missing_keys
    )
except:
    _pt_model = None

if _tf_model is None:
    print("_tf_model fails")
else:
    print("_tf_model OK")

if _pt_model is None:
    print("_pt_model fails")
else:
    print("_pt_model OK")
```

### CODE REVIEW

File: src/transformers/modeling_tf_pytorch_utils.py [hunk 1]
```diff
@@ -499,15 +511,27 @@ def load_tf2_state_dict_in_pytorch_model(pt_model, tf_state_dict, allow_missing_
...
+            _name = key_components[-2] + "_g"
+        elif key_components[-3::2] == ["parametrizations", "original1"]:
+            _name = key_components[-2] + "_v"
+        if _name is not None:
+            key_components = key_components[:-3] + [_name]
+            _pt_weight_name = ".".join(key_components)
+
         # Find associated numpy array in pytorch model state dict
-        if pt_weight_name not in tf_weights_map:
+        if _pt_weight_name not in tf_weights_map:
```
Comment (ydshieh) [inline src/transformers/modeling_tf_pytorch_utils.py:527]: This is used to check against the dict `tf_weights_map`.

File: src/transformers/modeling_tf_pytorch_utils.py [hunk 2]
```diff
@@ -273,6 +273,18 @@ def load_pytorch_state_dict_in_tf2_model(
...
+        key_components = key.split(".")
+        _name = None
+        if key_components[-3::2] == ["parametrizations", "original0"]:
+            _name = key_components[-2] + "_g"
+        elif key_components[-3::2] == ["parametrizations", "original1"]:
+            _name = key_components[-2] + "_v"
+        if _name is not None:
+            key_components = key_components[:-3] + [_name]
+            new_key = ".".join(key_components)
+
```
Comment (ydshieh) [inline src/transformers/modeling_tf_pytorch_utils.py:287]: Not able to put this in the method

```python
convert_tf_weight_name_to_pt_weight_name
```

File: src/transformers/modeling_tf_pytorch_utils.py [hunk 3]
```diff
@@ -273,6 +273,18 @@ def load_pytorch_state_dict_in_tf2_model(
             new_key = key.replace("running_var", "moving_variance")
         if "running_mean" in key:
             new_key = key.replace("running_mean", "moving_mean")
+
+        # New `weight_norm` from https://github.com/huggingface/transformers/pull/24030
+        key_components = key.split(".")
+        _name = None
```
Comment (sgugger) [inline src/transformers/modeling_tf_pytorch_utils.py:279]: Let's use another name if `name` is taken, but no leading underscore in variable names please.

File: src/transformers/modeling_tf_pytorch_utils.py [hunk 4]
```diff
@@ -499,15 +511,27 @@ def load_tf2_state_dict_in_pytorch_model(pt_model, tf_state_dict, allow_missing_
             new_pt_params_dict[pt_weight_name] = loaded_pt_weights_data_ptr[pt_weight.data_ptr()]
             continue
 
+        _pt_weight_name = pt_weight_name
```
Comment (sgugger) [inline src/transformers/modeling_tf_pytorch_utils.py:514]: Same here.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this! Just a few comments on the variable names.

**Review by sgugger — APPROVED** [review]

**Pull request merged**